src/main/java/Duke.java

```diff
@@ -1,10 +1,128 @@
+import java.util.Scanner;
+import java.util.Arrays;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
+    private static void logoWithHello() {
+        String logo = " ____        _\n"
+                + "|  _ \\ _   _| | _____\n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
     }
+
+    public static void horizontalLine() {
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void greet() {
+        horizontalLine();
+        System.out.println("Hello! I'm Duke\nWhat can I do for you?\n");
+        horizontalLine();
+    }
+
+    public static void exit() {
+        horizontalLine();
+        System.out.println("Bye. Hope to see you again soon!\n");
+        horizontalLine();
+    }
+
+    public static void echo(String userInput) {
+        horizontalLine();
+        System.out.println("added: " + userInput);
+        horizontalLine();
+    }
+
+    private static void listCommand(Task[] tasks) {
+        horizontalLine();
+        for (int i = 0; i < Task.getNumberOfTasks(); i += 1) {
+            System.out.println(tasks[i].getTaskNumber() + "." + tasks[i]);
+        }
+        horizontalLine();
+    }
+
+    public static String[] processLine(String userInput) {
+        String[] words = userInput.split(" ");
+        return words;
+    }
+
+    public static void taskAdded(Task t) {
+        horizontalLine();
+        boolean lessThanOne = (Task.getNumberOfTasks() <= 1);
+        System.out.println("Got it. I've added this task:\n" + t.toString());
+        System.out.println("Now you have " + Task.getNumberOfTasks() + (lessThanOne ? " task" : " tasks") + " in the list\n");
+        horizontalLine();
+    }
+
+    public static void processCommands(Task[] tasks, String userInput) {
+        String[] words = processLine(userInput);
+        if (words[0].equals("list")) {
+            listCommand(tasks);
+        } else if (words[0].equals("todo")) {
+            todoCommand(tasks, userInput);
+        } else if (words[0].equals("event")) {
+            eventCommand(tasks, userInput);
+        } else if (words[0].equals("deadline")) {
+            deadlineCommand(tasks, userInput);
+        } else if (words[0].contains("mark")) {
+            markUnmarkCommand(tasks, words);
+        }
+    }
+
+    private static void todoCommand(Task[] tasks, String userInput) {
+        userInput = userInput.substring(5);
```

Comment:
Would you like to consider using a named constant instead of having a magic number as it might cause confusion why you would choose the number 5? The same applies for the rest of the class

```suggestion
        int lengthOfTo = "/todo".length;
        userInput = userInput.substring(lengthOfTo);
```